### DISCUSSION TITLE
How to get a lexicon file from spaCy ?

### DISCUSSION THREAD

**Discussion by jacklanda** [explosion/spaCy#11805] [Help: Coding & Implementations] [answered]
There is a issue now, I have processed multiple texts using the same pipeline. It's known that the lexicon in pipeline is available for each text which was input in. How can I get the lexicon (maybe a dictionary ?) like the format below to the disk. Thanks !

<img width="471" alt="图片" src="https://user-images.githubusercontent.com/54089835/201822200-71ab01e0-6496-4661-b127-2d0051dfcd43.png">

**Comment by polm** [comment, marked as answer]
Please don't post screenshots of text, post text as text, for example using [a markdown code block](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code). 

I am not sure what you mean by "lexicon", but you can produce output similar to your example like this:

```
import spacy
nlp = spacy.load("en_core_web_sm")

for tok in nlp("This is input"):
    print(tok.i, tok, tok.lemma_, tok.pos_, sep="\t")
```

You can redirect that to a file in the shell, or you can save it to a string and write a file as usual. 

Does that answer your question?

**Reply by jacklanda** [reply]
> Please don't post screenshots of text, post text as text, for example using [a markdown code block](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code).
> 
> I am not sure what you mean by "lexicon", but you can produce output similar to your example like this:
> 
> ```
> import spacy
> nlp = spacy.load("en_core_web_sm")
> 
> for tok in nlp("This is input"):
>     print(tok.i, tok, tok.lemma_, tok.pos_, sep="\t")
> ```
> 
> You can redirect that to a file in the shell, or you can save it to a string and write a file as usual.
> 
> Does that answer your question?

This answer inspires me. Thanks so much!
